rancher2/data_source_rancher2_global_role.go

```diff
@@ -0,0 +1,43 @@
+package rancher2
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceRancher2GlobalRole() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceRancher2GlobalRoleRead,
+		Schema: globalRoleFields(),
```

Comment:
DataSource and Resource shouldn't share same schema as arguments properties should be different. All dataSource arguments but `name` should be computed.